x/wasm-storage/keeper/msg_server_test.go

```diff
@@ -28,32 +34,66 @@ func (s *KeeperTestSuite) TestStoreDataRequestWasm() {
 			preRun: func() {},
 			input: types.MsgStoreDataRequestWasm{
 				Sender:   s.authority,
-				Wasm:     compWasm,
+				Wasm:     regWasmZipped,
 				WasmType: types.WasmTypeDataRequest,
 			},
 			expErr: false,
 			expOutput: types.MsgStoreDataRequestWasmResponse{
-				Hash: hex.EncodeToString(crypto.Keccak256(wasm)),
+				Hash: hex.EncodeToString(crypto.Keccak256(regWasm)),
 			},
 		},
 		{
-			name: "Overlay wasm already exist",
+			name: "Data Request wasm already exist",
 			input: types.MsgStoreDataRequestWasm{
 				Sender:   s.authority,
-				Wasm:     compWasm,
+				Wasm:     regWasmZipped,
 				WasmType: types.WasmTypeDataRequest,
 			},
 			preRun: func() {
 				input := types.MsgStoreDataRequestWasm{
 					Sender:   s.authority,
-					Wasm:     compWasm,
+					Wasm:     regWasmZipped,
 					WasmType: types.WasmTypeDataRequest,
 				}
 				_, err := s.msgSrvr.StoreDataRequestWasm(s.ctx, &input)
 				s.Require().Nil(err)
 			},
 			expErr:    true,
-			expErrMsg: "Data Request Wasm with given hash already exists",
+			expErrMsg: "data Request Wasm with given hash already exists",
+		},
+		// TO-DO: Add after migrating ValidateBasic logic
```

Comment:
Why commented?

Is the logic not already there?
https://github.com/sedaprotocol/seda-chain/blob/6a1f50d7e9e0187817f466fa3c15406037fb7ff5/x/wasm-storage/types/tx.go#L22

Reply:
Yes! But Cosmos is encouraging us to put that logic directly in `msg_server` starting with v0.50. See https://github.com/cosmos/cosmos-sdk/issues/15648